### DISCUSSION TITLE
Absolute values

### DISCUSSION THREAD

**Discussion by infinite-pursuits** [Consensys/gnark#909] [Q&A]
Hi,
I am trying to write a circuit to do checks of the kind Ax = b. Due to rounding errors in converting the variables to integers, the values are only approximately equal. So I need to check if Ax - b is a small value, which can be done with api.AssertIsLessOrEqual(Ax-b, limit). However, I do not know if Ax > b or b>Ax --> so sometimes the api call works other times it fails. One way to circumvent this issue is to square the terms and call api.AssertIsLessOrEqual((Ax-b)^2, limit^2).

However, is there a better way to do this? I basically need absolute values.

**Comment by infinite-pursuits** [comment]
Also is there a general rule of thumb for setting the value of limit?

**Reply by ivokub** [reply]
Have you looked at [std/math/cmp](https://pkg.go.dev/github.com/consensys/gnark@v0.9.1/std/math/cmp#BoundedComparator) package? I think it could help you.

Otherwise the approach would be to check that (Ax-b)+limit is "small" (less than 2*limit).
